scrapy/extensions/robotstxtparser.py

```diff
@@ -0,0 +1,52 @@
+from six.moves.urllib_robotparser import RobotFileParser
+
+from scrapy.utils.python import to_native_str
+
+class BaseRobotsTxtParser():
+    def __init__(self, robotstxt_url):
+        raise NotImplementedError
+
+    def parse(self, robotstxt_body):
+        """Parse content of robots.txt file."""
+        raise NotImplementedError
+
+    def allowed(self, url, useragent):
+        """Return True if url is allowed for crawling, otherwise return False."""
+        raise NotImplementedError
+
+    def sitemaps(self):
```

Comment:
How will Scrapy use this method?

If it will use it, I wonder if it would make sense to make `sitemaps` return an iterator.

Reply:
Currently, Scrapy does not use `sitemaps`. Sitemap is a part of robot inclusion protocol and provides a list of URLs available for crawling on the website. I thought, Scrapy can make use of this list as a starting point for crawling the website. This could be useful if say, some pages are disconnected (no other page links to them but that is very unlikely to happen). But, it seems to me that sitemaps is rarely used. I am unable to make an opinion on whether to include it in the interface or not.

@whalebot-helmsman what's your opinion on this?

Reply:
Well, we do have a SitemapSpider class. I’ve just looked at the [SitemapSpider documentation](https://docs.scrapy.org/en/latest/topics/spiders.html#sitemapspider), and it looks like it already supports reading sitemap URLs from a `robots.txt` file. Maybe, as a stretch goal, we could look into getting SitemapSpider to use the configured `robots.txt` parse.

So yes, +1 to `sitemaps()`, but I do think an iterator may be more flexible. So, instead of returning a list, the method could yield the sitemap URLs.